### DISCUSSION TITLE
Is square missing in price_to_tick function?

### DISCUSSION THREAD

**Discussion by nikoulai** [Jeiwan/uniswapv3-book#73] [Milestone 1. First Swap]
In the tick calculations the formula i = log<sub>√1.0001</sub>(p(i)) but in the [implementation](https://github.com/Jeiwan/uniswapv3-code/blob/0924bbd239cb153a9975a83c5d376f42e839f457/unimath.py#L11C26-L11C26) of price_to_tick, square root is not present. Is it missing or I miss something?

**Comment by Jeiwan** [comment]
No need to take the square root in this case since the function takes a price, not the square root of a price.

The formula is:
$$i = log_{\sqrt{1.0001}} \sqrt{P(i)}$$

It's derived from:
$$\sqrt{P(i)}=1.0001^{\frac{i}{2}}$$

If you cancel out the square root on the both sides you'll get the formula from the implementation.
